### ISSUE TITLE
Positioning, color, and styling of buttons in Settings Editor are inconsistent with how buttons are used in jupyterlab

### ISSUE THREAD

**Issue by andrii-i** [jupyterlab/jupyterlab#14009]
## Description

Settings Editor UI: buttonbar positioning (left v right usually in jupyterlab) and styling (absence of spacing between buttons, absence of spacing between button bar and section border) and color palette (Move Up and Move Down are similar actions but button colors are inconsistent: grey and black) are inconsistent with use and styling of buttons in other places in jupyterlab

## Reproduce

1. Open Settings Editor
2. Open various accordions. For example, cell toolbar
3. Note positioning, styling, and colors of buttons

## Expected behavior

Buttons are positioned at bottom right part of the form, their styling is consistent with how buttons are usually used in jupyterlab

![image](https://user-images.githubusercontent.com/26686070/218910483-bcdf022f-6b3a-404d-81ee-7fb00d95cfb4.png)

## Context

Operating System and version: MacOS 12.6.3
Browser and version: Chrome version 109.0.5414.87 (Official Build) (arm64)
JupyterLab version: head-of-master (WIP 4.0.0)

**Comment by krassowski** [comment]
> Buttons are positioned at bottom right part of the form

There are some other places where these appear, e.g. if the list is more compact:

![Screenshot from 2023-02-15 21-04-27](https://user-images.githubusercontent.com/5832902/219160441-edd5b6f2-2282-490b-ade2-193226a03e33.png)

On related point, should we use icons instead of verbose titles? Or a combination of both?

**Comment by andrii-i** [comment]
That's a good question @krassowski. Jupyterlab UI as it is now seems to favor having verbose titles wherever possible.  Various toolbars are only UI elements where icon + title is used. 

It would be nice to formalize jupyter / jupyterlab UI language into guidelines at some point where such questions would be explicitly answered.

**Comment by GabrielaVives** [comment]
@krassowski , @andrii-i , I agree that icons+tooltip would be better than these repetitive buttons. Probably on the top right, aligned with each section title. Something like this:

<img width="1568" alt="image" src="https://user-images.githubusercontent.com/105775162/228797360-cf361f6b-ea0d-45ab-b30d-521f609e8637.png">

**Comment by andrii-i** [comment]
Here's a preview of a Settings Editor rework I'm working on
![new_settings_editor](https://user-images.githubusercontent.com/26686070/228918182-62ea87a4-7d39-4294-ba72-4e26aab16483.gif)
